libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/DropInNavigationView.kt

```diff
@@ -81,12 +80,14 @@ class DropInNavigationView @JvmOverloads constructor(
     }
 
     init {
-        val style = if (isNightModeEnabled()) {
-            navigationContext.options.mapStyleUriNight.value
-        } else {
-            navigationContext.options.mapStyleUriDay.value
-        }
-        binding.mapView.getMapboxMap().loadStyleUri(style)
+        lifecycle.addObserver(
+            MapStyleLoader(
+                context,
+                binding.mapView.getMapboxMap(),
+                navigationContext.options,
+                navigationContext.loadedMapStyle
+            )
```

Comment:
Did you consider adding the `MapStyleLoader` instance to the `DropInNavigtationViewContext`

I think we should expose a class for downstream components to use. Instead of exposing the MutableStateFlow

Reply:
I did consider and initially tried approach. Unfortunately it lead to massive memory leak somewhere that took way too much time to hunt down. 
So I moved  'loadedMapStyle' StateFlow outside and refactored MapStyleLoader to ensure that it is only retained by the Lifecycle.

Reply:
Oh interesting.. Do we need to null out the `DropInNavigationViewContext` when the view is destroyed? Do you know where the leak is happening? I can help hunt it down